### ISSUE TITLE
Invoke_shell self refrences

### ISSUE THREAD

**Issue by Kildar1988** [paramiko/paramiko#1754]
In the file client.py def invoke_shell calls its self before the return creating a forever loop.

**Comment by ploxiln** [comment]
No, that's not what it does ... [SSHClient.invoke_shell()](https://github.com/paramiko/paramiko/blob/2.7.2/paramiko/client.py#L520) calls [Channel.invoke_shell()](https://github.com/paramiko/paramiko/blob/2.7.2/paramiko/channel.py#L206) ... if you encountered a problem then it must be caused by something else.
